tests/test_references.py

```diff
@@ -224,3 +227,38 @@ def test_external_references() -> None:
     output, unmapped = fix_refs(source, url_map.__getitem__)
     assert output == '<a class="autorefs autorefs-external" href="https://example.com">example</a>'
     assert unmapped == []
+
+
+def test_register_markdown_anchors() -> None:
+    """Check that Markdown anchors are registered when enabled."""
+    plugin = AutorefsPlugin()
+    md = markdown.Markdown(extensions=["attr_list", AutorefsExtension(plugin)])
+    plugin.current_page = ""
+    md.convert(
+        dedent(
+            """
+            [](){#foo}
+            ## Heading foo
+
+            Paragraph 1.
+
+            [](){#bar}
+            Paragraph 2.
+
+            [](){#alias1}
+            [](){#alias2}
+            ## Heading bar
+
+            [](){#alias3}
+            Text.
+            [](){#alias4}
+            ## Heading baz
+            """,
+        ),
+    )
+    assert plugin._url_map["foo"] == "#heading-foo"
+    assert plugin._url_map["bar"] == "#bar"
+    assert plugin._url_map["alias1"] == "#heading-bar"
+    assert plugin._url_map["alias2"] == "#heading-bar"
+    assert plugin._url_map["alias3"] == "#alias3"
+    assert plugin._url_map["alias4"] == "#heading-baz"
```

Comment:
Would be nice to have an official specification of what is expected to happen in case of conflicts, with the most extreme case being this one:

```markdown
[](){#foo}
## Bar

[](){#bar}
## Foo
```

Reply:
With 

```md
[](){#foo}
## Bar
...
[](){#bar}
## Foo
...

[Link to foo][foo]
[Link to bar][bar]
```

![foobar](https://github.com/mkdocstrings/autorefs/assets/3999221/2f16c909-1838-4085-a2bb-edca51c15432)

Erm sorry about the highlighted rectangle on the left, but you get the idea.

Clicking on the "foo" link will bring you to the "Bar" heading, and inversely.